src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -285,4 +338,51 @@ public void indexAnomalyResultException(
         }
     }
 
+    private void confirmTotalRCFUpdatesFound(
+        String detectorId,
+        String taskState,
+        Long rcfTotalUpdates,
+        Long detectorIntervalInMinutes,
+        String error,
+        ActionListener<Long> listener
+    ) {
+        nodeStateManager.getAnomalyDetector(detectorId, ActionListener.wrap(detectorOptional -> {
+            if (!detectorOptional.isPresent()) {
+                listener.onFailure(new AnomalyDetectionException(detectorId, "fail to get detector"));
+                return;
+            }
+            nodeStateManager.getAnomalyDetectorJob(detectorId, ActionListener.wrap(jobOptional -> {
+                if (!jobOptional.isPresent()) {
+                    listener.onFailure(new AnomalyDetectionException(detectorId, "fail to get job"));
+                    return;
+                }
+                adTaskCacheManager.markResultIndexQueried(detectorId);
+                ProfileUtil
+                    .confirmDetectorInitStatus(
+                        detectorOptional.get(),
+                        jobOptional.get().getEnabledTime().toEpochMilli(),
+                        client,
+                        ActionListener.wrap(searchResponse -> {
+                            ActionListener.completeWith(listener, () -> {
+                                SearchHits hits = searchResponse.getHits();
+                                Long correctedTotalUpdates = rcfTotalUpdates;
+                                if (hits.getTotalHits().value > 0L) {
+                                    // correct the number if we have already had results after job enabling time
+                                    // so that the detector won't stay initialized
+                                    correctedTotalUpdates = Long.valueOf(rcfMinSamples);
```

Comment:
are we just going to manually set number of updates to 32 if there are some anomaly results even if technically it could be higher than 32 cause we have no real way of knowing the real number when cache is cleared?

Reply:
yes, you are right